superset/views/dashboard/api.py

```diff
@@ -0,0 +1,281 @@
+            self.instance.owners.append(g.user)
+        for field in data:
+            if field == "owners":
+                self.set_owners(self.instance, data["owners"])
+            else:
+                setattr(self.instance, field, data.get(field))
+        for slc in self.instance.slices:
+            slc.owners = list(set(self.instance.owners) | set(slc.owners))
+        return self.instance
+
+
+class DashboardRestApi(DashboardMixin, ModelRestApi):
+    datamodel = SQLAInterface(models.Dashboard)
+
+    resource_name = "dashboard"
+    allow_browser_login = True
+
+    class_permission_name = "Dashboard"
+    method_permission_name = {
+        "get_list": "list",
+        "get": "show",
+        "post": "add",
+        "put": "edit",
+        "delete": "delete",
+        "info": "list",
+    }
+    exclude_route_methods = ("info",)
+    show_columns = [
+        "dashboard_title",
+        "slug",
+        "owners.id",
+        "owners.username",
+        "position_json",
+        "css",
+        "json_metadata",
+        "published",
+        "table_names",
+        "charts",
+    ]
+
+    add_model_schema = DashboardPostSchema()
+    edit_model_schema = DashboardPutSchema()
+
+    @expose("/", methods=["POST"])
+    @protect()
+    @safe
+    def post(self):
+        """Creates a new dashboard
+        ---
+        post:
+          requestBody:
+            description: Model schema
+            required: true
+            content:
+              application/json:
+                schema:
+                  $ref: '#/components/schemas/{{self.__class__.__name__}}.post'
+          responses:
+            201:
+              description: Dashboard added
+              content:
+                application/json:
+                  schema:
+                    type: object
+                    properties:
+                      id:
+                        type: string
+                      result:
+                        $ref: '#/components/schemas/{{self.__class__.__name__}}.post'
+            400:
+              $ref: '#/components/responses/400'
+            401:
+              $ref: '#/components/responses/401'
+            422:
+              $ref: '#/components/responses/422'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        if not request.is_json:
+            return self.response_400(message="Request is not JSON")
+        item = self.add_model_schema.load(request.json)
+        # This validates custom Schema with custom validations
+        if item.errors:
+            return self.response_422(message=item.errors)
+        try:
+            self.datamodel.add(item.data, raise_exception=True)
+            return self.response(
+                201,
+                **{
+                    "result": self.add_model_schema.dump(item.data, many=False).data,
+                    "id": self.datamodel.get_pk_value(item.data),
```

Comment:
I think it would be slightly easier to read this if we instead captured the result of `self.datamodel.add(item.data, raise_exception=True)` and then piped in the `id` from that